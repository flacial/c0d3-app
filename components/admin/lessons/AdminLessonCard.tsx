import React from 'react'
import Link from 'next/link'
import styles from '../../../scss/adminLessonCard.module.scss'
import { Lesson } from '../../../graphql/'

type Props = {
  lesson: Lesson
  pendingFlaggedQuestions: number
}

export const AdminLessonCard: React.FC<Props> = ({
  lesson,
  pendingFlaggedQuestions
}) => {
  const { title, description, docUrl } = lesson

  return (
    <div className={styles.container}>
      <div className={styles.container__heading}>{title}</div>

      <div className={styles.container__description}>{description}</div>
      <div className={styles.container__flaggedQuestions}>
        {pendingFlaggedQuestions} Pending Questions
      </div>

      <Link href={docUrl || '_blank'}>
        <a data-testid="button">
          <button className={'btn btn-secondary ' + styles.container__button}>
            Edit Lesson
          </button>
        </a>
      </Link>
    </div>
  )
}

export default AdminLessonCard
