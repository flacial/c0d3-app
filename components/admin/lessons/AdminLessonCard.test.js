import * as React from 'react'
import { render } from '@testing-library/react'
import { AdminLessonCard } from './AdminLessonCard'
import '@testing-library/jest-dom'

const dummyLessonData = {
  lesson: {
    title: 'Foundations of Javascript',
    description: 'A super simple introduction to help you get started!',
    challenges: [],
    id: 0,
    order: 0,
    slug: 'js0'
  },
  pendingFlaggedQuestions: 2
}

describe('AdminLessonCard component', () => {
  test("Button href should be '_blank' if lesson docUrl is undefined ", async () => {
    expect.assertions(2)
    const { container, getByTestId } = render(
      <AdminLessonCard {...dummyLessonData} />
    )

    expect(container).toMatchSnapshot()
    expect(getByTestId('button').href.split('/').slice(-1)[0]).toBe('_blank')
  })
})
