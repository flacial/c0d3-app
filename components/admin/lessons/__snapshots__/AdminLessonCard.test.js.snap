// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLessonCard component Button href should be '_blank' if lesson docUrl is undefined  1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container__heading"
    >
      Foundations of Javascript
    </div>
    <div
      class="container__description"
    >
      A super simple introduction to help you get started!
    </div>
    <div
      class="container__flaggedQuestions"
    >
      2
       Pending Questions
    </div>
    <a
      data-testid="button"
      href="/_blank"
    >
      <button
        class="btn btn-secondary container__button"
      >
        Edit Lesson
      </button>
    </a>
  </div>
</div>
`;
